lib/evss/ppiu/payment_information.rb

```diff
@@ -26,6 +26,16 @@ class PaymentInformation
       attribute :payment_account, EVSS::PPIU::PaymentAccount
       attribute :payment_address, EVSS::PPIU::PaymentAddress
       attribute :payment_type, String
+
+      delegate :authorized?, to: :control_information
+
+      def payment_account
+        authorized? ? @payment_account : EVSS::PPIU::PaymentAccount.new
+      end
+
+      def payment_address
+        authorized? ? @payment_address : EVSS::PPIU::PaymentAddress.new
+      end
```

Comment:
Can you help me understand why this is here? 

This part looks to be doing more than only restrict access to the `ppiu#update` endpoint. Are you also trying to filter out data from a response from some other endpoint as well?

Reply:
Also probably worth having a `payment_information_spec.rb` now that there's some logic in it.

Reply:
yes we also want to prevent unauthorized users from seeing the information in payment account and payment address

Reply:
thanks for the suggestion, i added `payment_information_spec.rb`